L1Trigger/CSCTrackFinder/src/CSCTFPtLUT.cc

```diff
@@ -86,16 +86,17 @@ CSCTFPtLUT::CSCTFPtLUT(const edm::EventSetup& es)
         //std::cout << "pt_method from 4 " << std::endl; 
 	lowQualityFlag = 4;
 	isBeamStartConf = true;
-	pt_lut = new ptdat[1<<21];
-
-	edm::ESHandle<L1MuCSCPtLut> ptLUT;
-	es.get<L1MuCSCPtLutRcd>().get(ptLUT);
-	const L1MuCSCPtLut *myConfigPt_ = ptLUT.product();
-
-	memcpy((void*)pt_lut,(void*)myConfigPt_->lut(),(1<<21)*sizeof(ptdat));
-
-	lut_read_in = true;
-
+	if ( !lut_read_in) {
+	  pt_lut = new ptdat[1<<21];
```

Comment:
even better is to remove the copy altogether and use the conditions object…but less trivial

> On Jun 25, 2015, at 3:22 PM, Chris Jones <notifications@github.com> wrote:
> 
> In L1Trigger/CSCTrackFinder/src/CSCTFPtLUT.cc:
> 
> > @@ -86,16 +86,17 @@ CSCTFPtLUT::CSCTFPtLUT(const edm::EventSetup& es)
> >          //std::cout << "pt_method from 4 " << std::endl; 
> >  	lowQualityFlag = 4;
> >  	isBeamStartConf = true;
> > -	pt_lut = new ptdat[1<<21];
> > -
> > -	edm::ESHandle<L1MuCSCPtLut> ptLUT;
> > -	es.get<L1MuCSCPtLutRcd>().get(ptLUT);
> > -	const L1MuCSCPtLut *myConfigPt_ = ptLUT.product();
> > -
> > -	memcpy((void*)pt_lut,(void*)myConfigPt_->lut(),(1<<21)*sizeof(ptdat));
> > -
> > -	lut_read_in = true;
> > -
> > +	if ( !lut_read_in) {
> > +	  pt_lut = new ptdat[1<<21];
> 
> Also better to use std::unique_ptr< T[]> to manage the memory.
> 
> —
> Reply to this email directly or view it on GitHub.
>